### PR TITLE
the alonzo UTxO rule to use alonzo minfee function

### PR DESCRIPTION
**Pull request by JaredCorduan**
This PR cherry picks the commit from #2936 for the release branch `release/1.0.0`. The cherry pick required a bit of massaging due to the glorious recent cleanup in #2901, but can be compared line by line with #2936 to check that is is semantically the same.

### CODE REVIEW

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs [hunk 1]
```diff
@@ -272,8 +271,10 @@ feesOK ::
     Core.Tx era ~ Alonzo.ValidatedTx era,
     -- "collateral" to get inputs to pay the fees
     HasField "collateral" (Core.TxBody era) (Set (TxIn (Crypto era))),
+    HasField "txrdmrs" (Core.Witnesses era) (Redeemers era),
```
Comment (teodanciu) [inline eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs:274]: This seems to be missing from the original PR..but probably not relevant

**Review by teodanciu — APPROVED** [review]
Diffing the diffs, I noticed that this PR adds  `HasField "txrdmrs" (Core.Witnesses era) (Redeemers era)` to `feesOK` function, whereas #2936 didn't. 
Also, in #2936,  `HasField "_prices" ` was added as constraints to `utxoTransition` and `  STS (AlonzoUTXO era)` instance, whereas in this PR we're only adding it to `feesOK`.   Perhaps these are necessary because of the other changes that were left behind on this branch!  
The rest looks the same to me :+1:

File: eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs (hunk 1 above)
Comment (JaredCorduan) [reply eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxo.hs:274]: yea, this was made obsolete by https://github.com/input-output-hk/cardano-ledger/pull/2901

**Comment by JaredCorduan** [comment]
I am going to merge, even though `ci/hydra:Cardano:cardano-ledger-specs:required` is reporting `Pending`. If you follow the link, all the tests have actually passed.

**Pull request merged**